### PR TITLE
Prevent chef-client launchd from unloading itself and add Catalina workaround

### PR DESCRIPTION
**Pull request by jazaval**
### Description
This PR resolves a macOS issue when attribute changes to the `/Library/LaunchDaemons/com.chef.chef-client.plist` are made during a chef-client converge initiated from that same file. The current `macosx_service` `:restart` implementation runs `launchctl unload` on the daemon, but cannot proceed to `load` it as macOS will terminate the converge process as a side effect of the `unload`. This results in a state where some external process must `launchctl load` the chef-client daemon again to resume interval runs. 

This PR also resolves an issue in macOS Catalina where even the root user (launchd default) cannot write to `/` (made read-only in 10.15). macOS uses `/` as a default launchd working directory, which causes [converge issues](https://github.com/chef/chef/issues/9338) in resources that use this folder when installing gems. A new attribute exposes this plist key/value and sets a functional default of `/var/root`. 
### Issues Resolved

https://github.com/chef/chef/issues/9338
### Check List

- [ ✅] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ✅] New functionality includes testing.
- [ ✅] New functionality has been documented in the README if applicable
- [ ✅] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

Signed-off-by: Jacob Zaval jazava@microsoft.com

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,7 +77,10 @@ The following attributes are set on a per-platform basis, see the `attributes/de
...
 This cookbook makes use of attribute-driven configuration with this attribute. See [USAGE](#usage) for examples.
-- `node['chef_client']['launchd_mode']` - (Only for Mac OS X) if set to `'daemon'`, runs chef-client with `-d` and `-s` options; defaults to `'interval'`.
+- `node['chef_client']['launchd_mode']` - (Only for Mac OS X) If set to `'daemon'`, runs chef-client with `-d` and `-s` options; defaults to `'interval'`.
+- `node['chef_client']['launchd_working_dir']` - (Only for Mac OS X) Sets the working directory for the launchd user (generally `root`); defaults to `/var/root`.
+- `node['chef_client']['launchd_self-update']` - (Only for Mac OS X) Determines whether chef-client should attempt to `:restart` itself when changes are made to the launchd plist during converge. Note that the [current implementation](https://github.com/chef/chef/blob/129a6f3982218e5eadcd33b272ba738b317bbcae/lib/chef/provider/service/macosx.rb#L128) of
```
Comment (americanhanko) [inline README.md]: Maybe?

```diff
-(Only for Mac OS X)
+(Only for macOS)
```

Comment (jazaval) [reply README.md]: Sure

**Comment by dheerajd-msys** [comment]
@jazaval Please check for the conflicts and rebase.

Thanks!

**Comment by dheerajd-msys** [comment]
@jazaval Please check ^

**Comment by jazaval** [comment]
@dheerajd-msys updated my branch with changes from master.

**Comment by jazaval** [comment]
@tas50 I'm getting some unit testing failures for `chef-client::cron` even after merging in `master`.

**Comment by tas50** [comment]
Thanks @jazaval

**Pull request merged**